File: Models/Ride.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SemCopilot.Models
{

    public class Ride
    {
        public string Id { get; set; }

        public string VehicleId { get; set; }
        public Vehicle Vehicle { get; set; }

        public string RiderId { get; set; }
        public User Rider { get; set; }

        public DateTime Date { get; set; }
    }
}

File: models/ride.js
const { Model, DataTypes } = require('sequelize');
const sequelize = require('../config/database'); // Adjust the path as needed

class Ride extends Model {}

Ride.init({
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
    allowNull: false
  },
  rider: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'user', // Name of the User table
      key: 'id'
    }
  },
  vehicle: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'vehicle', // Name of the Vehicle table
      key: 'id'
    }
  },
  date: {
    type: DataTypes.DATEONLY,
    allowNull: false
  },
}, {
  sequelize,
  modelName: 'Ride',
  tableName: 'ride',
  timestamps: false
});

module.exports = Ride;